sale_order_type/views/res_partner_view.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+        <record model="ir.ui.view" id="res_partner_sale_type_form_view">
+            <field name="name">res.partner.sale_type.form</field>
+            <field name="model">res.partner</field>
+            <field name="inherit_id" ref="base.view_partner_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//page[@name='sales_purchases']//field[@name='customer']"
```

Comment:
This is not the best place for this field. Put it in the sale group, not after the customer flag.